internal/indexcoord/meta_table.go

```diff
@@ -560,70 +667,55 @@ func (mt *metaTable) HasSameReq(req *indexpb.BuildIndexRequest) (bool, UniqueID)
 		if notEq {
 			continue
 		}
+		if meta.indexMeta.MarkDeleted {
+			continue
+		}
 		return true, meta.indexMeta.IndexBuildID
 	}
 
 	return false, 0
 }
 
-// LoadMetaFromETCD load the meta of specified indexBuildID from ETCD.
-// If the version of meta in memory is greater equal to the version in ETCD, no need to reload.
-func (mt *metaTable) LoadMetaFromETCD(indexBuildID int64, indexVersion int64) bool {
+// NeedUpdateMeta update the meta of specified indexBuildID.
+// If the version of meta in memory is greater equal to the version in put event, no need to update.
+func (mt *metaTable) NeedUpdateMeta(m *Meta) bool {
 	mt.lock.Lock()
 	defer mt.lock.Unlock()
-	meta, ok := mt.indexBuildID2Meta[indexBuildID]
-	log.Debug("IndexCoord metaTable LoadMetaFromETCD", zap.Int64("indexBuildID", indexBuildID),
-		zap.Int64("indexVersion", indexVersion), zap.Bool("ok", ok))
-	if ok {
-		log.Debug("IndexCoord metaTable LoadMetaFromETCD",
-			zap.Int64("meta.indexVersion", meta.indexVersion),
-			zap.Int64("indexVersion", indexVersion))
 
-		if meta.indexVersion >= indexVersion {
-			return false
-		}
-	} else {
-		log.Error("Index not exist", zap.Int64("IndexBuildID", indexBuildID))
-		return false
-	}
-
-	m, err := mt.reloadMeta(indexBuildID)
-	if m == nil {
-		log.Error("IndexCoord metaTable reloadMeta failed", zap.Error(err))
+	meta, ok := mt.indexBuildID2Meta[m.indexMeta.IndexBuildID]
+	if !ok {
+		log.Warn("index is not exist, Might have been cleaned up meta", zap.Int64("buildID", m.indexMeta.IndexBuildID))
 		return false
 	}
+	log.Info("IndexCoord metaTable NeedUpdateMeta", zap.Int64("indexBuildID", m.indexMeta.IndexBuildID),
+		zap.Int64("meta.revision", meta.etcdVersion), zap.Int64("update revision", m.etcdVersion))
 
-	mt.indexBuildID2Meta[indexBuildID] = *m
-	log.Debug("IndexCoord LoadMetaFromETCD success", zap.Any("IndexMeta", m))
-
-	return true
-}
-
-// GetNodeTaskStats get task stats of IndexNode.
-func (mt *metaTable) GetNodeTaskStats() map[UniqueID]int {
-	mt.lock.RLock()
-	defer mt.lock.RUnlock()
-
-	log.Debug("IndexCoord MetaTable GetPriorityForNodeID")
-	nodePriority := make(map[UniqueID]int)
-	for _, meta := range mt.indexBuildID2Meta {
-		if meta.indexMeta.State == commonpb.IndexState_InProgress {
-			nodePriority[meta.indexMeta.NodeID]++
-		}
+	if meta.etcdVersion < m.etcdVersion {
+		mt.indexBuildID2Meta[m.indexMeta.IndexBuildID] = m
+		return true
 	}
-	return nodePriority
+	return false
 }
 
-// GetIndexMetaByIndexBuildID get the index meta of the specified indexBuildID.
-func (mt *metaTable) GetIndexMetaByIndexBuildID(indexBuildID UniqueID) *indexpb.IndexMeta {
+func (mt *metaTable) CanBeRecycled(buildID UniqueID) bool {
```

Comment:
It is recommended to change the name to CanRecycleIndexFiles.

Reply:
done